File: integration/tools/send_to_agent.py
"""send_to_agent: encrypt+ship a message (optionally with a Bitcoin payment)."""

from __future__ import annotations

from communication.channel.agent_channel import AgentChannel
from integration.rpc.methods import RpcMethod
from integration.rpc.schema import SendToAgentParams, SendToAgentResult


class SendToAgentMethod(RpcMethod):
    """JSON-RPC method that calls AgentChannel.send."""

    name = "send_to_agent"
    params_schema = SendToAgentParams
    result_schema = SendToAgentResult

    async def __call__(
        self,
        channel: AgentChannel,
        params: SendToAgentParams,
    ) -> SendToAgentResult:
        # Convert BTCPayloadDTO → BTCPayload, await channel.send, wrap MessageId in result.
        ...

File: skills/openclaw-trustroom/src/tools/send_to_agent.ts
// send_to_agent: encrypt + ship a message (optionally with a Bitcoin payment).

import type { JSONRpcClient } from "../client.js";

export interface SendToAgentParams {
    room_id: string;
    text: string;
    payment?: {
        amount_sats: number;
        recipient_btc_pubkey_hex: string;
        signed_tx_hex: string;
    };
}

export interface SendToAgentResult {
    message_id: string;
}

export const sendToAgent = {
    name: "send_to_agent",
    description: "Send a message (optionally with a Bitcoin payment) to an agent in a Trustroom.",
    handler: async (
        params: SendToAgentParams,
        ctx: { client: JSONRpcClient },
    ): Promise<SendToAgentResult> => {
        // Forward to the Python sidecar via JSONRpcClient.call("send_to_agent", params).
        return ctx.client.call<SendToAgentResult>("send_to_agent", params);
    },
};
